src/WidgetBase.ts

```diff
@@ -71,6 +69,23 @@ export function diffProperty(propertyName: string, diffType = DiffType.CUSTOM, d
 }
 
 /**
+ * Decorator that can be used to enable property diffing of all properties
+ */
+export function diffProperties(constructor: Function): void;
```

Comment:
However with the changes for enhanced decorators we are going to change all decorators to accept an optional function - https://github.com/dojo/widget-core/issues/402